src/compiler/checker.ts

```diff
@@ -6925,7 +6966,11 @@ namespace ts {
             checkCollisionWithCapturedThisVariable(node, node);
             checkBlockScopedBindingCapturedInLoop(node, symbol);
 
-            return getNarrowedTypeOfSymbol(getExportSymbolOfValueSymbolIfExported(symbol), node);
+            const type = getNarrowedTypeOfSymbol(getExportSymbolOfValueSymbolIfExported(symbol), node);
+            if (type === undefinedType || type == nullType) {
+                return (contextualMapper || identityMapper)(type);

```

Comment:
@ahejlsberg can correct me, but I think the contextualMapper is just for inferential typing.